geopandas/plotting.py

```diff
@@ -693,7 +696,16 @@ def plot_dataframe(
             fmt = "{:.2f}"
             if legend_kwds is not None and "fmt" in legend_kwds:
                 fmt = legend_kwds.pop("fmt")
+
             categories = binning.get_legend_classes(fmt)
+            show_interval = True
+            if legend_kwds is not None and "interval" in legend_kwds:
+                show_interval = legend_kwds.pop("interval")
+                if not show_interval:
+                    categories = [
+                        c.replace("(", "").replace(")", "").replace("[", "").replace("]", "")
+                        for c in categories
+                    ]
```

Comment:
```suggestion
                    categories = [c[1:-1] for c in categories]
```

This would be easier.